cloud/deployment/fromfile/fromfile.go

```diff
@@ -158,189 +148,416 @@ func CreateOrUpdate(inputFile, action string, client astro.Client, coreClient as
 	var (
-		defaultOptions astro.WorkerQueueDefaultOptions
-		configOptions  astro.DeploymentConfig
-		listQueues     []astro.WorkerQueue
-		astroMachine   astro.Machine
-		createInput    astro.CreateDeploymentInput
-		updateInput    astro.UpdateDeploymentInput
-		err            error
+		defaultOptions          astroplatformcore.WorkerQueueOptions
+		configOptions           astroplatformcore.DeploymentOptions
+		listQueues              []astroplatformcore.WorkerQueue
+		astroMachine            astroplatformcore.WorkerMachine
+		listQueuesRequest       []astroplatformcore.WorkerQueueRequest
+		listHybridQueuesRequest []astroplatformcore.HybridWorkerQueueRequest
+		err                     error
 	)
+	deploymentType := transformDeploymentType(deploymentFromFile.Deployment.Configuration.DeploymentType)
 
 	// Add worker queues if they were requested
 	if hasQueues(deploymentFromFile) {
 		// transform inspect.WorkerQ to []astro.WorkerQueue
-		listQueues, err = getQueues(deploymentFromFile, nodePools, existingDeployment.WorkerQueues)
+		if existingDeployment.WorkerQueues != nil {
+			listQueues = *existingDeployment.WorkerQueues
+		}
+		listQueues, err = getQueues(deploymentFromFile, nodePools, listQueues)
+		if err != nil {
+			return err
+		}
+		// get defaults for min-count, max-count and concurrency from API
+		deploymentOptions, err := deployment.GetPlatformDeploymentOptions("", astroplatformcore.GetDeploymentOptionsParams{}, astroPlatformCore)
 		if err != nil {
-			return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{}, err
+			return err
 		}
-		if deploymentFromFile.Deployment.Configuration.Executor == deployment.CeleryExecutor {
-			// get defaults for min-count, max-count and concurrency from API
-			defaultOptions, err = workerqueue.GetWorkerQueueDefaultOptions(client)
-			if err != nil {
-				return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{}, err
-			}
-			if deployment.IsDeploymentHosted(deploymentFromFile.Deployment.Configuration.DeploymentType) || deployment.IsDeploymentDedicated(deploymentFromFile.Deployment.Configuration.DeploymentType) {
-				configOptions, err = client.GetDeploymentConfig()
-				if err != nil {
-					return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{}, err
+
+		defaultOptions = deploymentOptions.WorkerQueues
+		configOptions = deploymentOptions
+		for i := range listQueues {
+			if deployment.IsDeploymentStandard(deploymentType) || deployment.IsDeploymentDedicated(deploymentType) {
+				astroMachines := configOptions.WorkerMachines
+				for j := range astroMachines {
+					if astroMachines[j].Name == *listQueues[i].AstroMachine {
```

Comment:
does this work instead?
```suggestion
					if astroMachines[j].Name == listQueues[i].AstroMachine {
```

Reply:
astroMachines[j].Name needs to be a string